### DISCUSSION TITLE
cfg(target_arch) in build scripts for FFI still refer to the host rather than the target

### DISCUSSION THREAD

**Discussion by jguhlin** [cross-rs/cross#1285] [Q&A] [answered]
Running
```
cross +nightly test --target aarch64-unknown-linux-gnu
```

Will trigger this part of the build script:
```
#[cfg(target_arch = "x86_64")]
fn target_specific(cc: &mut cc::Build) {
    println!("Building for x86_64");
```

Any ideas? I need to set up target specific optimizations or the software can get incredibly slow.

**Comment by Emilgardis** [comment, marked as answer]
`cfg(target_arch)` will always refer to the target the current crate is built for, and build scripts are always built for the host, not the target.

Instead, use the environment variable [`CARGO_CFG_TARGET_ARCH`](https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts)

as noted in the text, ensure that you actually retrieve the env var at runtime using `env::var("CARGO_CFG_TARGET_ARCH")`

note that this behaviour is not changed by cross, rather it's how cargo/rust works

**Comment by jguhlin** [comment]
Thanks, this is helpful!
